ompi/mca/coll/ucc/coll_ucc_allgatherv.c

```diff
@@ -0,0 +1,111 @@
+
+/**
+ * Copyright (c) 2021 Mellanox Technologies. All rights reserved.
+ * $COPYRIGHT$
+ *
+ * Additional copyrights may follow
+ *
+ */
+
+#include "coll_ucc_common.h"
+
+static inline ucc_status_t mca_coll_ucc_allgatherv_init(const void *sbuf, int scount,
+                                                        struct ompi_datatype_t *sdtype,
+                                                        void* rbuf, const int *rcounts, const int *rdisps,
+                                                        struct ompi_datatype_t *rdtype,
+                                                        mca_coll_ucc_module_t *ucc_module,
+                                                        ucc_coll_req_h *req,
+                                                        mca_coll_ucc_req_t *coll_req)
+{
+    ucc_datatype_t         ucc_sdt, ucc_rdt;
+
+    ucc_sdt = ompi_dtype_to_ucc_dtype(sdtype);
+    ucc_rdt = ompi_dtype_to_ucc_dtype(rdtype);
+    if (COLL_UCC_DT_UNSUPPORTED == ucc_sdt ||
+        COLL_UCC_DT_UNSUPPORTED == ucc_rdt) {
+        UCC_VERBOSE(5, "ompi_datatype is not supported: dtype = %s",
+                    (COLL_UCC_DT_UNSUPPORTED == ucc_sdt) ?
+                    sdtype->super.name : rdtype->super.name);
+        goto fallback;
+    }
+
+    ucc_coll_args_t coll = {
+        .mask      = UCC_COLL_ARGS_FIELD_FLAGS,
+        .coll_type = UCC_COLL_TYPE_ALLGATHERV,
+        .flags = UCC_COLL_ARGS_FLAG_CONTIG_SRC_BUFFER |
```

Comment:
how can you assume it is that the buffers are CONTIG? i think it is not correct unless you perform O(N) check.